src/components/dataDisplays/DataDisplay.jsx

```diff
@@ -159,36 +161,45 @@ export default function DataDisplay({
                   .filter(c =>
                     get(c, 'options.displayInFilter', true),
                   )
-                  .map(c => (
-                    <FormControlLabel
-                      key={c.name}
-                      control={
-                        <Checkbox
-                          size="small"
-                          checked={visibleColumnNames.includes(
-                            c.name,
-                          )}
-                          onClick={() => {
-                            if (visibleColumnNames.includes(c.name)) {
-                              if (visibleColumnNames.length === 1)
-                                return;
-                              setVisibleColumnNames(
-                                visibleColumnNames.filter(
-                                  vc => vc !== c.name,
-                                ),
-                              );
-                            } else {
-                              setVisibleColumnNames([
-                                ...visibleColumnNames,
-                                c.name,
-                              ]);
-                            }
-                          }}
-                        />
-                      }
-                      label={<Text variant="body2">{c.label}</Text>}
-                    />
-                  ))}
+                  .map(c => {
+                    const columnLabel = c?.labelId
+                      ? intl.formatMessage({ id: c.labelId })
+                      : c.label;
+                    return (
+                      <FormControlLabel
+                        key={c.name}
+                        control={
+                          <Checkbox
+                            size="small"
+                            checked={visibleColumnNames.includes(
+                              c.name,
+                            )}
+                            onClick={() => {
+                              if (
+                                visibleColumnNames.includes(c.name)
+                              ) {
+                                if (visibleColumnNames.length === 1)
```

Comment:
Why wouldn't we want a table with just one visible column to return anything?

Reply:
So actually the only thing that changed here was the indentation. But this prevents people from hiding the last column of a table when they are showing/hiding columns.